src/intel_pmu.c

```diff
@@ -283,6 +285,27 @@ static void pmu_submit_counter(int cpu, char *event, counter_t value) {
   plugin_dispatch_values(&vl);
 }
 
+meta_data_t *pmu_event_get_meta(struct event *e, int cpu) {
```

Comment:
it would be great to rename this function to pmu_event_set_meta rather than get... but other than that this looks good to me

Reply:
Typically we call constructors (functions that allocate, initialize and return a `foo_t`) `foo_create`. The `pmu` prefix is fine, which would make this `pmu_meta_data_create`.